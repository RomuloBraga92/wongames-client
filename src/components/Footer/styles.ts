import styled, { css } from 'styled-components'
import media from 'styled-media-query'

import * as HeadingStyles from 'components/Heading/styles'

export const Container = styled.footer`
  ${HeadingStyles.Container} {
    text-transform: uppercase;
  }
`

export const Content = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-column-gap: ${theme.grid.gutter};
    margin-top: ${theme.spacings.medium};

    ${media.greaterThan('medium')`
      grid-template-columns: repeat(4, 1fr);
    `}
  `}
`

export const Column = styled.div`
  ${({ theme }) => css`
    a,
    span {
      display: block;
      text-decoration: none;
      color: ${theme.colors.gray};
      margin-bottom: ${theme.spacings.xxsmall};
      font-size: ${theme.font.sizes.small};
    }

    a:hover {
      text-decoration: underline;
    }
  `}
`

export const Copyright = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    margin-top: ${theme.spacings.large};
    margin-bottom: ${theme.spacings.medium};
    font-size: ${theme.font.sizes.xsmall};
    text-align: center;
  `}
`
