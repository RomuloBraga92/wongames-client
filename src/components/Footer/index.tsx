import * as S from './styles'

const Footer = () => (
  <S.Container>
    <h1>Footer</h1>
  </S.Container>
)

export default Footer
