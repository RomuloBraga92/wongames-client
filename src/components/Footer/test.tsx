import { render, screen } from '@testing-library/react'

import Footer from './'

describe('<Footer />', () => {
  it('should render the heading', () => {
    const { container } = render(<Footer />)

    expect(screen.getByRole('heading', { name: /Footer/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
