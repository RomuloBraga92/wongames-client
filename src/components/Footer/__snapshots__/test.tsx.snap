// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    Footer
  </h1>
</main>
`;
